packages/workers-shared/asset-server-worker/src/index.ts

```diff
@@ -1,38 +1,62 @@
+import { WorkerEntrypoint } from "cloudflare:workers";
 import { AssetsManifest } from "./assets-manifest";
+import {
+	InternalServerErrorResponse,
+	MethodNotAllowedResponse,
+	NotFoundResponse,
+	OkResponse,
+} from "./responses";
+import { getAdditionalHeaders, getMergedHeaders } from "./utils/headers";
+import { getAssetWithMetadataFromKV } from "./utils/kv";
 
-interface Env {
-	/**
-	 * ASSETS_MANIFEST is a pipeline binding to an ArrayBuffer containing the
-	 * binary-encoded site manifest
-	 */
-	ASSETS_MANIFEST: ArrayBuffer;
-	/**
-	 * ASSETS_KV_NAMESPACE is a pipeline binding to the KV namespace that the
-	 *  assets are in.
-	 */
-	ASSETS_KV_NAMESPACE: KVNamespace;
-}
-
-export default {
-	async fetch(request: Request, env: Env) {
-		const { ASSETS_MANIFEST, ASSETS_KV_NAMESPACE } = env;
+export default class extends WorkerEntrypoint<Env> {
+	async fetch(request: Request) {
+		if (request.method.toLowerCase() !== "get") {
```

Comment:
todo @CarmenPopoviciu add fixture test for this

Reply:
added [here](https://github.com/cloudflare/workers-sdk/pull/6539/commits/98ab8e48594cd6de89127e95794b061cd0621380)